packages/components/src/Interactive.ts

```diff
@@ -35,124 +35,133 @@ const allowedHoverVariables = [
   'boxShadow',
   'opacity',
   'textDecoration',
-];
+] as const;
+
+// Union type of all values in allowedHoverVariables.
+type HoverStyleProperty = (typeof allowedHoverVariables)[number];
```

Comment:
`number` here is an _indexing type_ used get the type of `allowedHoverVariables`' elements. 
https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html

Since `allowedHoverVariables` is declared using a _const assertion_ (`as const`), the element type is more specific than `string` and contains the exact values listed.
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-inference